code/modules/mob/living/simple_animal/hostile/mining_mobs/elites/elite.dm

```diff
@@ -0,0 +1,352 @@
+	background_icon_state = "bg_default"
+	var/mob/living/simple_animal/hostile/asteroid/elite/M
+	var/chosen_message
+	var/chosen_attack_num = 0
+
+/datum/action/innate/elite_attack/Grant(mob/living/L)
+	if(istype(L, /mob/living/simple_animal/hostile/asteroid/elite))
+		M = L
+		return ..()
+	return FALSE
+
+/datum/action/innate/elite_attack/Activate()
+	M.chosen_attack = chosen_attack_num
+	to_chat(M, chosen_message)
+	
+/mob/living/simple_animal/hostile/asteroid/elite/updatehealth()
+	. = ..()
+	update_health_hud()
+		
+/mob/living/simple_animal/hostile/asteroid/elite/update_health_hud()
+	if(hud_used)
+		var/severity = 0
+		var/healthpercent = (health/maxHealth) * 100
+		switch(healthpercent)
+			if(100 to INFINITY)
+				hud_used.healths.icon_state = "elite_health0"
+			if(80 to 100)
+				hud_used.healths.icon_state = "elite_health1"
+				severity = 1
+			if(60 to 80)
+				hud_used.healths.icon_state = "elite_health2"
+				severity = 2
+			if(40 to 60)
+				hud_used.healths.icon_state = "elite_health3"
+				severity = 3
+			if(20 to 40)
+				hud_used.healths.icon_state = "elite_health4"
+				severity = 4
+			if(10 to 20)
+				hud_used.healths.icon_state = "elite_health5"
+				severity = 5
+			if(1 to 20)
+				hud_used.healths.icon_state = "elite_health6"
+				severity = 6
+			else
+				hud_used.healths.icon_state = "elite_health7"
+				severity = 7
+		if(severity > 0)
+			overlay_fullscreen("brute", /obj/screen/fullscreen/brute, severity)
+		else
+			clear_fullscreen("brute")
+
+//The Pulsing Tumor, the actual "spawn-point" of elites, handles the spawning, arena, and procs for dealing with basic scenarios.
+
+/obj/structure/elite_tumor
+	name = "pulsing tumor"
+	desc = "An odd, pulsing tumor sticking out of the ground.  You fell compelled to reach out and touch it..."
+	armor = list("melee" = 100, "bullet" = 100, "laser" = 100, "energy" = 100, "bomb" = 100, "bio" = 100, "rad" = 100, "fire" = 100, "acid" = 100)
+	resistance_flags = INDESTRUCTIBLE
+	var/activity = TUMOR_INACTIVE
+	var/boosted = FALSE
+	var/times_won = 0
+	var/mob/living/carbon/human/activator = null
+	var/mob/living/simple_animal/hostile/asteroid/elite/mychild = null
+	var/potentialspawns = list(/mob/living/simple_animal/hostile/asteroid/elite/broodmother,
+								/mob/living/simple_animal/hostile/asteroid/elite/pandora,
+								/mob/living/simple_animal/hostile/asteroid/elite/legionnaire,
+								/mob/living/simple_animal/hostile/asteroid/elite/herald)
+	icon = 'icons/obj/lavaland/tumor.dmi'
+	icon_state = "tumor"
+	pixel_x = -16
+	light_color = LIGHT_COLOR_RED
+	light_range = 3
+	anchored = TRUE
+	density = FALSE
+	
+/obj/structure/elite_tumor/attack_hand(mob/user)
+	. = ..()
+	if(ishuman(user))
+		switch(activity)
+			if(TUMOR_INACTIVE)
+				activity = TUMOR_ACTIVE
+				var/mob/dead/observer/elitemind = null
+				visible_message("<span class='boldwarning'>The tumor begins to convulse.  Your instincts tell you to step back.</span>")
+				activator = user
+				if(boosted)
```

Comment:
This would read much nicer if you did.!boosted then add the timer and return. Then you don't have to nest so deep